### ISSUE TITLE
Feature request: ANSI color output in doctests

### ISSUE THREAD

**Issue by mnemnion** [JuliaDocs/Documenter.jl#2432]
This is certainly just a nice-to-have, but it would be nice. 

My use case: I'm working on a parser generator, and the `PegFail` struct returned from a failed parse highlights the character at which the parse failed in red. Doctests ignore this when comparing the output (which is very nice) so it's possible to write examples containing this type of return value as doctests. But then the output isn't highlighted in the resulting documentation. 

My workaround has been to use a \@repl block, which does provide the color output, but now I don't get notified if there's a regression. The example will always be up-to-date, but if I want to make sure it's still _correct_ I have to check that myself. 

Coloring the output would, I feel, be the most consistent choice, since the other styles of block already do so, and that keeps it consistent with the use of `ansicolor=false|true` everywhere else.

**Comment by mortenpi** [comment]
Do doctests with ANSI codes work in the REPL? Although, even if they do, they would not be very readable in the source code. I agree that it would look great in the rendered documentation, but I'm unsure about how to reconcile it with the rest of the doctest architecture.

**Comment by mnemnion** [comment]
The doctests work, yes. And the way they work is rather nice: the ANSI codes are ignored, so colored output is compared to the plain text output. I assume this is using an `IOContext` with `color=false`. 

What I'm proposing is to keep that behavior, just color the actual rendered HTML. There isn't any good way to include the colors in the "answer" part of the test anyway, so I think the part it's already doing is the right thing, and then going on to color the actual rendered output should be harmless, and generally what the user would prefer, with `ansicolor=false` available if they don't

**Comment by mortenpi** [comment]
> There isn't any good way to include the colors in the "answer" part of the test anyway, so I think the part it's already doing is the right thing, and then going on to color the actual rendered output should be harmless

Documenter doesn't evaluate the doctests when it's rendering them (just when it checks them). So in the current architecture, the color information needs to go into the `jldoctest` blocks.

**Comment by mortenpi** [comment]
Yeah, I think technically re-executing all doctests as if they're at-repl blocks basically, would be doable (although it might involve some refactoring of the doctest code). But the semantics get weird. E.g. what would you do with doctest filters?

I'd say we could try some opt-in experiment, but it's not 100% obvious to me what this change should look like.

**Comment by mnemnion** [comment]
That raises a good point. I haven't used doctest filters, but I just gave the docs a fresh read, and it doesn't look like anything would need to change in terms of how they function. 

If I'm understanding the feature correctly, doctests are run on both the output in the `.md` and the output from running the code.  Then when the docs are rendered, the version from the `.md` file is used, without any filtering being applied.  This is already being done on uncolored output, and since nothing in this proposal changes the doctests, nothing can break there. 

So if there's a random number being generated, the HTML will contain the exact number given in the documentation.  So this feature would change behavior, since if Documenter is now using the output from running the code, the number would be whatever it happened to be while building the docs.  This could definitely lead to unexpected results, for instance, that number would be different building locally and building remotely. 

It's unclear to me whether this should be considered a breaking change. I'm sure Documenter makes no guarantees that the HTML will stay identical, that's far too restrictive to be practical. But this is a genuine behavior change, and it's easy for me to imagine it being unwanted. For instance, a user might have made that random number an injoke, their zipcode, `1337`, or something like that, and wouldn't appreciate Documenter suddenly changing that number to be truly random. 

So I would suggest that it should be considered breaking, and introduced on an opt-in, experimental basis. My sense is that the Documenter maintainers are reaching the point where a 2.0 will be forthcoming, and that would be an opportunity to reverse the default and make "your contents will be the result of executing the code, including color" the opt-out, rather than the opt-in. I don't personally have a strong opinion on which default is better, actually, just which one I would choose.
